controllers/provider-alicloud/pkg/webhook/controlplane/ensurer.go

```diff
@@ -116,3 +118,13 @@ func (e *ensurer) EnsureKubeletConfiguration(ctx context.Context, kubeletConfig
 	kubeletConfig.FeatureGates["KubeletPluginsWatcher"] = true
 	return nil
 }
+
+// EnsureAdditionalUnits ensures that additional required system units are added.
+func (e *ensurer) EnsureAdditionalUnits(ctx context.Context, units *[]extensionsv1alpha1.Unit) error {
```

Comment:
Instead of implementing `EnsureAdditionalUnits` and `EnsureAdditionalFiles` this for every individual `ensurer` please implement this here: https://github.com/gardener/gardener-extensions/blob/master/pkg/webhook/controlplane/genericmutator/noopensurer.go

Then you can save all this code and don't need to define the method at all for all the different webhook ensurers that don't want to implement it.